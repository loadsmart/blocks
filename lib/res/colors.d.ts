declare const Colors: {
    AlgaeGreen: string;
    BrownGreyThree: string;
    Charcoal: string;
    CoralPink: string;
    DarkGrey: string;
    OrangePeel: string;
    LightGray: string;
    Tuna: string;
    VeryLightGray: string;
    White: string;
};
export default Colors;
