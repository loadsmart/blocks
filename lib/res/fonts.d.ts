declare const Fonts: {
    SharpSansBold: string;
    SharpSansItalic: string;
    SharpSansBook: string;
    SharpSansBookItalic: string;
    SharpSansExtrabold: string;
    SharpSansExtraboldItalic: string;
    SharpSansLight: string;
    SharpSansLightItalic: string;
    SharpSansMedium: string;
    SharpSansMediumItalic: string;
    SharpSansSemibold: string;
    SharpSansSemiboldItalic: string;
    SharpSansThin: string;
    SharpSansThinItalic: string;
};
export default Fonts;
