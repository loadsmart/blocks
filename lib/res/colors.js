"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Colors = {
    AlgaeGreen: '#14d64d',
    BrownGreyThree: '#9b9b9b',
    Charcoal: '#3c3d3a',
    CoralPink: '#fd576a',
    DarkGrey: '#666666',
    OrangePeel: '#ff9800',
    LightGray: '#d4d4d4',
    Tuna: '#484a4c',
    VeryLightGray: '#eee',
    White: '#fff',
};
exports.default = Colors;
