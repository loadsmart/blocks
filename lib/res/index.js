"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = __importDefault(require("./colors"));
exports.Colors = colors_1.default;
var fonts_1 = __importDefault(require("./fonts"));
exports.Fonts = fonts_1.default;
var images_1 = __importDefault(require("./images"));
exports.Images = images_1.default;
