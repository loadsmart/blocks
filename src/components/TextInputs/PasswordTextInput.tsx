import React, { Component } from 'react'
import {
  Image,
  StyleSheet,
  TextInput as RNTextInput,
  TextInputProps,
  TouchableOpacity,
  View,
} from 'react-native'
import { Colors, Fonts, Images } from '../../res'
import { ThemeContext } from '../Contexts'

interface Props extends TextInputProps {
  focusable?: boolean
}

interface State {
  focused: boolean
  passwordVisible: boolean
}

export default class PasswordTextInput extends Component<Props, State> {
  public static defaultProps: Props = {
    focusable: true,
  }

  constructor(props: Props, state: State) {
    super(props, state)
    this.state = {
      focused: false,
      passwordVisible: false,
    }
  }

  public render() {
    return (
      <ThemeContext.Consumer>
        {theme => {
          const borderStyle =
            this.props.focusable === true && this.state.focused === true
              ? { borderColor: theme.primaryColor }
              : { borderColor: Colors.White }
          return (
            <View style={[this.props, styles.container, this.props.style, borderStyle]}>
              <RNTextInput
                style={styles.textInput}
                onFocus={this.onFocus.bind(this)}
                onBlur={this.onBlur.bind(this)}
                secureTextEntry={!this.state.passwordVisible}
                onChangeText={this.props.onChangeText}
              />
              <TouchableOpacity
                onPress={() => this.setState({ passwordVisible: !this.state.passwordVisible })}
              >
                <Image
                  style={styles.icon}
                  source={this.state.passwordVisible ? Images.PasswordShow : Images.PasswordHide}
                />
              </TouchableOpacity>
            </View>
          )
        }}
      </ThemeContext.Consumer>
    )
  }

  public onFocus() {
    this.setState(_ => ({
      focused: true,
    }))
  }

  public onBlur() {
    this.setState(_ => ({
      focused: false,
    }))
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 52,
    backgroundColor: Colors.White,
    borderRadius: 26,
    paddingHorizontal: 28,
    fontFamily: Fonts.SharpSansSemibold,
    fontSize: 16,
    color: Colors.Charcoal,
    borderWidth: 2,
  },
  textInput: {
    flex: 1,
    alignSelf: 'flex-start',
    alignItems: 'center',
    height: '100%',
    padding: 5,
  },
  icon: {
    flex: 1,
    alignSelf: 'flex-end',
    alignItems: 'center',
    resizeMode: 'contain',
    width: 25,
    height: 25,
    padding: 10,
    backgroundColor: 'transparent',
  },
})
