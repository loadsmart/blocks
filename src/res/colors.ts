const Colors = {
  AlgaeGreen: '#14d64d',
  BrownGreyThree: '#9b9b9b',
  Charcoal: '#3c3d3a',
  CoralPink: '#fd576a',
  DarkGrey: '#666666',
  OrangePeel: '#ff9800',
  LightGray: '#d4d4d4',
  Tuna: '#484a4c',
  VeryLightGray: '#eee',
  White: '#fff',
  Ghost: '#c4ccd6',
  Transparent: '#00000000',
}

export default Colors
