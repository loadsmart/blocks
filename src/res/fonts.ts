const Fonts = {
  SharpSansBold: 'SharpSans-Bold',
  SharpSansItalic: 'SharpSans-BoldItalic',
  SharpSansBook: 'SharpSans-Book',
  SharpSansBookItalic: 'SharpSans-BookItalic',
  SharpSansExtrabold: 'SharpSans-Extrabold',
  SharpSansExtraboldItalic: 'SharpSans-ExtraboldItalic',
  SharpSansLight: 'SharpSans-Light',
  SharpSansLightItalic: 'SharpSans-LightItalic',
  SharpSansMedium: 'SharpSans-Medium',
  SharpSansMediumItalic: 'SharpSans-MediumItalic',
  SharpSansSemibold: 'SharpSans-Semibold',
  SharpSansSemiboldItalic: 'SharpSans-SemiboldItalic',
  SharpSansThin: 'SharpSans-Thin',
  SharpSansThinItalic: 'SharpSans-ThinItalic',
}

export default Fonts
