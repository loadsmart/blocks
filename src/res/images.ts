const Images = {
  Chevron: require('../../assets/images/chevron.png'),
  PlusSign: require(`../../assets/images/plus-sign.png`),
  SmallCheck: require(`../../assets/images/small-check.png`),
  Warning: require(`../../assets/images/warning.png`),
  PasswordShow: require('../../assets/images/password-show.png'),
  PasswordHide: require('../../assets/images/password-hide.png'),
}

export default Images
