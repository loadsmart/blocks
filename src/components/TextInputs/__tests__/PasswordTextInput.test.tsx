import React from 'react'
import { Image, TextInput as RNTextInput, TouchableOpacity } from 'react-native'
import renderer from 'react-test-renderer'
import { Images } from '../../../res'
import PasswordTextInput from '../PasswordTextInput'

describe('PasswordTextInput', () => {
  describe('when focusable is true', () => {
    it('renders with white border when on blur', () => {
      const testRenderer = renderer.create(<PasswordTextInput focusable />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onBlur()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('renders with green border when on focus', () => {
      const testRenderer = renderer.create(<PasswordTextInput focusable />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onFocus()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when focusable is false', () => {
    it('renders with white border when on blur', () => {
      const testRenderer = renderer.create(<PasswordTextInput focusable={false} />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onBlur()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('also renders with white border when on focus', () => {
      const testRenderer = renderer.create(<PasswordTextInput focusable={false} />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onFocus()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when password toggle is clicked', () => {
    it('renders password icons', () => {
      const testRenderer = renderer.create(<PasswordTextInput focusable={true} />)
      const toggle = testRenderer.root.findByType(TouchableOpacity)
      const image = testRenderer.root.findByType(Image)

      toggle.instance.props.onPress()
      expect(image.props.source).toBe(Images.PasswordShow)

      toggle.instance.props.onPress()
      expect(image.props.source).toBe(Images.PasswordHide)
    })
  })
})
