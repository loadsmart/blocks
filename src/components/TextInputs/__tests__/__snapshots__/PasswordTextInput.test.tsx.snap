// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextInput when focusable is false also renders with white border when on focus 1`] = `
<View
  style={
    Array [
      Object {
        "focusable": false,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "flexDirection": "row",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flex": 1,
        "height": "100%",
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/password-hide.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "transparent",
          "flex": 1,
          "height": 25,
          "padding": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </View>
</View>
`;

exports[`PasswordTextInput when focusable is false renders with white border when on blur 1`] = `
<View
  style={
    Array [
      Object {
        "focusable": false,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "flexDirection": "row",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flex": 1,
        "height": "100%",
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/password-hide.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "transparent",
          "flex": 1,
          "height": 25,
          "padding": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </View>
</View>
`;

exports[`PasswordTextInput when focusable is true renders with green border when on focus 1`] = `
<View
  style={
    Array [
      Object {
        "focusable": true,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "flexDirection": "row",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#14d64d",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flex": 1,
        "height": "100%",
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/password-hide.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "transparent",
          "flex": 1,
          "height": 25,
          "padding": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </View>
</View>
`;

exports[`PasswordTextInput when focusable is true renders with white border when on blur 1`] = `
<View
  style={
    Array [
      Object {
        "focusable": true,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "flexDirection": "row",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flex": 1,
        "height": "100%",
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/password-hide.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "transparent",
          "flex": 1,
          "height": 25,
          "padding": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
